assets/dev/js/editor/controls/media.js

```diff
@@ -222,27 +254,79 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		}
 
 		if ( attachment.url ) {
-			this.setValue( {
+			const updatedState = {
 				url: attachment.url,
 				id: attachment.id,
 				alt: attachment.alt,
 				source: attachment.source,
-			} );
+			};
+			this.setValue( updatedState );
 
 			if ( this.model.get( 'responsive' ) ) {
 				// Render is already calls `applySavedValue`, therefore there's no need for it in this case.
 				this.renderWithChildren();
 			} else {
 				this.applySavedValue();
 			}
-		}
 
+			this.updatePrevState( updatedState );
+			this.toggleSizeControl();
+		}
 		this.trigger( 'after:select' );
 	},
 
 	onBeforeDestroy() {
 		this.$el.remove();
 	},
+
+	async onMediaImageSize() {
+		this.ui.mediaImageSize.removeClass( 'e-select-placeholder' );
+		const selectedImage = {
+			id: this.getEditSettings( 'mediaId' ),
+			alt: this.getEditSettings( 'mediaAlt' ),
+		};
+
+		const selectedSize = this.ui.mediaImageSize.val();
+		let imageURL = await elementor.imagesManager.getImageUrl( {
```

Comment:
```suggestion
		let imageURL = elementor.imagesManager.getImageUrl( {
```

(you can probably also remove the `async` from the function signature)

Reply:
Removed